adding dagProcessor logic to cluster activity page

erdos2n opened the pull request.

erdos2n committed:
first commit

erdos2n committed:
refactoring

erdos2n committed:
adding undefined default

erdos2n committed:
added logic to check if defined

bbovenzi commented on airflow/www/static/js/cluster-activity/live-metrics/Health.tsx:
```diff
@@ -81,7 +81,16 @@ const HealthSection = ({
   const { data, isError } = useHealth();
-
+  let dagProcessorHealthComponent = null;
+  if (standaloneDagProcessor != undefined && standaloneDagProcessor.toString().toLowerCase() === "true") {
```
Where is this variable coming from?

erdos2n replied:
the `standaloneDagProcessor` variable?

I defined it as a global variable here:
https://github.com/erdos2n/airflow/blob/0c8ef2e3967e3baacdf8f702d854cdf03cfdca06/airflow/www/static/js/index.d.ts#L25 

I grab the value in the `views.py` file

https://github.com/erdos2n/airflow/blob/0c8ef2e3967e3baacdf8f702d854cdf03cfdca06/airflow/www/views.py#L1054

bbovenzi commented on airflow/www/static/js/cluster-activity/live-metrics/Health.tsx:
```diff
@@ -107,11 +116,7 @@ const Health = (props: CenterProps) => {
-              status={data?.dagProcessor?.status}
-              latestHeartbeat={data?.dagProcessor?.latestDagProcessorHeartbeat}
-            />
+            {dagProcessorHealthComponent}
```
```suggestion
            {
            standaloneDagProcessor != undefined && standaloneDagProcessor.toString().toLowerCase() === "true" && (
              <HealthSection
                title="Dag Processor"
                status={data?.dagProcessor?.status}
                latestHeartbeat={data?.dagProcessor?.latestDagProcessorHeartbeat}
              />
 )}
```

Let's not make the component a variable, just add the logic in the jsx.

Taragolis committed:
Use `NOT EXISTS` subquery instead of `tuple_not_in_condition` (#33527)

* Use `NOT EXISTS` subquery instead of `tuple_not_in_condition`

* Remove deprication from `tuple_not_in_condition`

erdos2n committed:
refactored component

bhagany committed:
Add missing template fields to DataformCreateCompilationResultOperator (#33585)

All other operators template these fields, and `impersonation_chain` is explicity said to be templated in the operator's docstring

bbovenzi commented on airflow/www/static/js/cluster-activity/live-metrics/Health.tsx:
```diff
@@ -107,11 +106,16 @@ const Health = (props: CenterProps) => {
               mb={3}
             />
+            { 
+            standaloneDagProcessor != undefined && 
```
```suggestion
            standaloneDagProcessor !== undefined && 
```
typo

erdos2n commented on airflow/www/templates/airflow/cluster_activity.html:
```diff
@@ -43,6 +43,7 @@
   <script>
     const stateColors = {{ state_color_mapping|tojson }};
     const autoRefreshInterval = {{ auto_refresh_interval }};
+    const standaloneDagProcessor = {{ standalone_dag_processor }} ;
```
@bbovenzi if there is a way to make this a boolean value then it would be much better. Currently when I pass in a boolean from `views.py` it throws an error because Python's `True` is capitalized.

bbovenzi replied:
`const standaloneDagProcessor = {{ standalone_dag_processor }} === 'True';`
This will convert it to boolean

jedcunningham committed:
Fix security model link (#33597)

erdos2n commented on airflow/www/views.py:
```diff
@@ -1051,10 +1051,12 @@ def cluster_activity(self):
         """Cluster Activity view."""
         state_color_mapping = State.state_color.copy()
         state_color_mapping["no_status"] = state_color_mapping.pop(None)
+        standalone_dag_processor = str(conf.getboolean("scheduler", "standalone_dag_processor")).lower()
```
@bbovenzi I had to convert to string and lowercase this value, but I'm not sure if this covers all possible values for `standalone_dag_processor` like if it equals 0 or 1. Not sure how these types of situations with booleans are typically handled in the codebase.

bbovenzi replied:
Given other sppts where we pass this. I think we can just pass the regular value and js will interpret it as a string of "True" or "False"

erdos2n commented on airflow/www/static/js/index.d.ts:
```diff
@@ -22,6 +22,7 @@ import type { DepEdge, DepNode } from "./types";
 // define global variables that come from FAB
 declare global {
   const autoRefreshInterval: number | undefined;
+  const standaloneDagProcessor: string | undefined;
```
if we can use a bool, it is preferred, but I can't figure out how to do it.

vandonr-amz committed:
make `conf.set` case insensitive (#33452)

* make `conf.set` case insensitive

`conf.get` is insensitive (it converts section and key to lower case)
but set is not, which can lead to surprising behavior
(see the test, which is not passing without the fix).
I suggest that we override set as well to fix that.
Any value that was set before with upper case was unreacheable.

* fix remove_option as well

* away with the str()

* add significant change newsfragment

eumiro committed:
Use str.splitlines() to split lines in providers (#33593)

Taragolis committed:
Move `.coveragerc` content to `pyproject.toml` (#33589)

erdos2n committed:
refactored not equal sign

eumiro committed:
Refactor: Simplify code in Apache/Alibaba providers (#33227)

mati-o committed:
Make cluster_context templated (#33604)

potiuk committed:
Improve detection of when breeze CI image needs rebuilding (#33603)

* Improve detection of when breeze CI image needs rebuilding

Previously we have been using provider.yaml file modification as
a sign that the docker image needs rebuilding when starting image.
However just modification of provider.yaml file is not a sign
that the image needs rebuilding. The image needs rebuilding when
provider dependencies changed, but there are many more reasons why
provider.yaml file changed - especially recently provider.yaml
file contains much more information and dependencies are only part
of it. Provider.yaml files can also be modified by release manager
wnen documentation is prepared, but none of the documentation
change is a reason for rebuilding the image.

This PR optimize the check for image building introducing two
step process:

* first we check if provider.yaml files changed
* if they did, we regenerate provider dependencies by manully
  running the pre-commit script
* then provider_dependencies.json is used instead of all providers
  to determine if the image needs rebuilding

This has several nice side effects:

* the list of files that have been modified displayed to the
  user is potentially much smaller (no provider.yaml files)
* provider_dependencies.json is regenereated automatically when
  you run any breeze command, which means that you do not have
  to have pre-commit installed to regenerate it
* the notification "image needs rebuilding" will be printed less
  frequently to the user - only when it is really needed
* preparing provider documentation in CI will not trigger
  image rebuilding (which might occasionally fail in such case
  especially when we bring back a provider from long suspension
  like it happened in #33574

* Update dev/breeze/src/airflow_breeze/commands/developer_commands.py

potiuk committed:
Fix failing "ARM cache" build (#33609)

The new version of autossh installed yesterday when our AMI has
been refreshed is a little less verbose when printing AUTOSSH_LOG
entries. So far it produced entries for starting and running the
autossh - which was not intersting if nothing was wrong:

```
2023/08/21 06:33:27 autossh[31898]: starting ssh (count 1)
2023/08/21 06:33:27 autossh[31898]: ssh child pid is 31899
```

The new versio will not produce any output and wont even create
a file when nothing wrong happens and this caused our CI
to fail at the very last moment - after the ARM instance has
been stopped.

This PR won't fail when the autossh log is missing.

erdos2n committed:
refactored code for boolean type instead of string

erdos2n commented:
@bbovenzi just refactored the code to use a boolean type. Can you check it?

Taragolis committed:
Add random suffix to coverage reports (#33605)

* Add random suffix for coverage reports

* Minor tweaks: enable MySQL coverage, disable Helm, increase color range

erdos2n committed:
refactored global variable. refactored conditional

vandonr-amz committed:
Fix AWS appflow waiter (#33613)

vincbeck committed:
Add a fallback in case no first name and last name are set (#33617)

mobuchowski committed:
openlineage: finish user guide (#33610)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

hussein-awala committed:
Revert "respect soft_fail argument when ExternalTaskSensor runs in deferrable mode (#33196)" (#33458)

This reverts commit a1b5bdb25a6f9565ac5934a9a458e9b079ccf3ae.

eumiro committed:
Simplify conditions on len() (#33454)

s0neq committed:
Resume yandex provider (#33574)


---------

Co-authored-by: sofya generalova <sageneralova@yandex-team.ru>

vandonr-amz committed:
Add Appflow system test + improvements (#33614)

ephraimbuddy committed:
Automatically update the REVISIONS_HEADS_MAP in db.py (#33616)

* Automatically update the REVISIONS_HEADS_MAP in db.py

This has been the responsibility of the release manager and manually done.
In 2.7.0, I mistakenly added a wrong value for the revision which means that
anyone upgrading with `airflow db migrate --version 2.7.0` will end up having incorrect
migration.
This PR automates this and always updates it to the correct value through pre-commit

* fixup! Automatically update the REVISIONS_HEADS_MAP in db.py

* Account for missing versions in revision heads map

* Provide the full path of the DB file in message

* Make REVISION_HEADS_MAP private

* Only add from 2.0.0 up

erdos2n committed:
pre-commit and linting fixes

erdos2n committed:
updated jest-setup

erdos2n committed:
updated jest-setup

erdos2n committed:
precommit fix

eumiro committed:
Fix typos (double words and it's/its) (#33623)

uranusjr committed:
Limit boto3 to help resolution (#33639)

* Limit boto3 to help resolution

* Better boto package combination

ephraimbuddy committed:
Minor updates to the release doc (#33650)

potiuk committed:
Replace deprecated slack notification in provider.yaml with new one (#33643)

The Slack Notification listed in provider.yaml for slack has
been still using the deprecated version of it. Since the
provider verification of ours did not check neither notifications
nor warnings, it was not found before.

Found in the attempt of adding new verification in #33608 then
also detected by #33640 that improved existing verification.

potiuk committed:
Exclude deprecated "operators.kubernetes_pod" module from provider.yaml (#33641)

Found when implementing #33640

SamWheating committed:
Bind engine before attempting to drop archive tables (#33622)

yermalov-here committed:
Fix AWS Batch waiter failure state (#33656)

* Fix AWS Batch waiter failure state

* Add tests for AWS Batch batch_job_complete waiter

potiuk committed:
Addd triggers and transfers to provider_info.schema (#33654)

They were not added to the schema, even if they are appearing
already as they are automatically extracted from provider.yaml

Part of #29918

potiuk committed:
Upgrade botocore/aiobotocore minimum requirements (#33649)

Botocore has a very peculiar process of releasing new version
every day, which means that it gives `pip` hard time to figure
what will be the non-conflicting set of packages when we have
too low of a minium version set as requirement.

Since we had > 1.24 that means that `pip` had to consider
more than 340 versions for botocore, but also for related
mypy packages and also a number of aiobotocore packages when
resolving eager-upgrade.

We limit all the relevant packages to 1.28 as minimum version
now, and we should continue doing that regularly in the future.

LipuFei committed:
Fix 2.7.0 db migration job errors (#33652)

* Fix circular import from kubernetes_executor

* Do not validate SQL when initialising Sentry

---------

Co-authored-by: Lipu Fei <lipu.fei@kpn.com>

frazar committed:
Fix broken link to the "UPDATING.md" file (#33583)

The "UPDATING.md" file has been removed in v2.3.0 with commit 5b9bd9954bbc6ed68c217761d581f3af46269af9, and therefore it does not exist on the `main` branch. Instead, link to the file's latest version (_i.e._, v2.2.5).

eumiro committed:
Refactor: Better percentage formatting (#33595)

zinal committed:
add support for Yandex Dataproc cluster labels (#29811)

Co-authored-by: Maksim Zinal <mzinal@ru.ibm.com>

potiuk committed:
Improve provider verification pre-commit (#33640)

There were a numer of problems with the provider verification pre-commit
scripts:

* It missed verification of "notifications"
* It did not check if the classes or modules specified in provider yaml
  raised deprecation warnings
* The messages produced by the script when some discrepancies were found
  were pretty cryptic and it was difficult to guess what kind of action
  should be taken to fix the problem

This PR fixes all those problems:

* verification of notification is performed
* when importing all the classes and modules, check for the
  AirflowProviderDeprecationWarnings is done and treated as error
* The messages produced provide clear actionable instructions on what
  to do and explain what are the discrepancies of expected vs. current
  list in a clear way

pankajastro committed:
Bump azure-kusto-data>=4.1.0 (#33598)

vincbeck committed:
Add back `get_url_for_login` in security manager (#33660)

* Add back `get_url_for_login` in security manager

* Rename FAB_SECURITY_MANAGER_CLASS to SECURITY_MANAGER_CLASS

alexbegg committed:
Referencing correct hooks for Apache Pinot (#33601)

* Referencing correct hooks for Apache Pinot

The `provider.yaml` for `airflow.providers.apache.pinot` provider was referencing a non-existing hook class of `PinotHook`, causing a warning during `providers_manager`'s sanity check. This corrects it to list both hooks of this provider.

* Adding missing attributes for `PinotDbApiHook` and `PinotAdminHook`

potiuk committed:
Bring back wrong "hook-class-names" check (#33662)

Continuation of #33640 - there were one more wrong check in
the provider, that missed "hook-class-names" check

erdos2n committed:
Update Health.tsx

replacing bottom margin with a top margin

erdos2n committed:
Update Health.tsx

removed bottom margin from triggerer
